src/client.ts

```diff
@@ -707,6 +707,7 @@ export interface ClientParams {
   port: number
   username: string
   password: string
+  mechanism?: string
```

Comment:
Instead of declaring as generic string why don't declaring as:
```suggestion
  mechanism?: "PLAIN" | "EXTERNAL"
```